sdks/java/extensions/sbe/src/main/java/org/apache/beam/sdk/extensions/sbe/SbeSchema.java

```diff
@@ -0,0 +1,252 @@
+            ir.packageName(),
+            ir.namespaceName(),
+            ir.id(),
+            ir.version(),
+            ir.description(),
+            ir.semanticVersion(),
+            ir.byteOrder(),
+            ImmutableList.copyOf(ir.headerStructure().tokens()));
+
+    return new SbeSchema(SerializableIr.fromIr(copy), irOptions, sbeFields);
+  }
+
+  @VisibleForTesting
+  @Nullable
+  Ir getIr() {
+    return ir == null ? null : ir.ir();
+  }
+
+  @VisibleForTesting
+  @Nullable
+  IrOptions getIrOptions() {
+    return irOptions;
+  }
+
+  @VisibleForTesting
+  ImmutableList<SbeField> getSbeFields() {
+    return sbeFields;
+  }
+
+  /**
+   * Options for controlling what to do with unsigned types, specifically whether to use a higher
+   * bit count or, in the case of uint64, a string.
+   */
+  @AutoValue
+  public abstract static class UnsignedOptions implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    public abstract Boolean useMoreBitsForUint8();
+
+    public abstract Boolean useMoreBitsForUint16();
+
+    public abstract Boolean useMoreBitsForUint32();
+
+    public abstract Boolean useStringForUint64();
+
+    /**
+     * Returns options for using the same bit size for all unsigned types.
+     *
+     * <p>This means that if an unsigned value from SBE comes in with a value outside the signed
+     * range, then the negative equivalent (in terms of bits) will be used.
+     */
+    public static UnsignedOptions usingSameBitSize() {
+      return UnsignedOptions.builder()
+          .setUseMoreBitsForUint8(false)
+          .setUseMoreBitsForUint16(false)
+          .setUseMoreBitsForUint32(false)
+          .setUseStringForUint64(false)
+          .build();
+    }
+
+    /**
+     * Returns options for using a higher bit count for unsigned types.
+     *
+     * <p>This means that if an unsigned value is encountered, it will always use the higher bit
+     * count, even if that higher bit count is unnecessary. However, this means that if it is
+     * necessary, then the proper value will be returned rather than the negative equivalent (in
+     * terms of bits).
+     *
+     * <p>The {@code includeUint64} controls the behavior of 64-bit values, since no properly
+     * higher-bit-numeric type exists. If true, then this will be converted into a string that must
+     * be parsed if intended to be used as a number. Otherwise, it will still be a 64-bit type and
+     * may return negative values.
+     */
+    public static UnsignedOptions usingHigherBitSize(boolean includeUint64) {
+      return UnsignedOptions.builder()
+          .setUseMoreBitsForUint8(true)
+          .setUseMoreBitsForUint16(true)
+          .setUseMoreBitsForUint32(true)
+          .setUseStringForUint64(includeUint64)
+          .build();
+    }
+
+    public static Builder builder() {
+      return new AutoValue_SbeSchema_UnsignedOptions.Builder();
+    }
+
+    /** Builder for {@link UnsignedOptions}. */
```

Comment:
I think it would be more straightforward, and acceptable, if this were just a schema level option. Also, it's odd as it is, since it's defined here in SbeSchema, but only used in the SbeFieldOptions